### PR TITLE
Adding a guide for contributors 

### PR DESCRIPTION
**Pull request by ethanshar**
## Description
Adding a guide for contributors 

## Changelog
Adding a guide for contributors

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+4. We try to make the PR review process as quick as possible, please be patient
+
+
+### Code Standards
+
+- Readability and clean code above all - we believe a clean code is easier to maintain and refactor when needed. 
+- Deciding on an API is not trivial but we always aim to keep it generic and intuitive - feel free to consult with us.
+- Our documentation is derived from our components' prop comments, make sure to add a clear description when adding new props. 
+- When possible, consider adding tets for a new functionality you add.
+
```
Comment (mendyEdri) [inline CONTRIBUTING.md:24]: 1. Maybe worth adding controlled / uncontrolled preference.
2. It's possible always to have tests 🧪, at least unit (obviously not mandatory).
3. Clean code - means to separate the ui from the logic? worth adding?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Contributing
+If you got here it means you want to dedicate some of your time and help us with making this library better.
```
Comment (Inbal-Tish) [inline CONTRIBUTING.md:2]:
```suggestion
If you got here it means you want to dedicate some of your time and help us improve this library.  
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# Contributing
+If you got here it means you want to dedicate some of your time and help us with making this library better.  
+So first, thank you ❤️
```
Comment (Inbal-Tish) [inline CONTRIBUTING.md:3]:
```suggestion
So first of all, thank you ❤️
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+When contributing to this repository, please first discuss the change you wish to make via [issue](https://github.com/wix/react-native-ui-lib/issues/new/choose),
+[Discord channel](https://discord.gg/2eW4g6Z), or any other method with the owners of this repository before making a change. 
+
+## Pull Request Process
+
+1. First make sure the environment is working and synced with master. For installation details go [here](https://github.com/wix/react-native-ui-lib/blob/master/markdowns/getting-started/setup.md#demo-app)
+2. Before submitting a PR we suggest running `npm run prepush` command that verifies our lint, TS and tests were not broken.
+3. Please don't change our PR template.
+- 3.1 In Description add everything that can help the reviewer and help the reviewing process.
```
Comment (Inbal-Tish) [inline CONTRIBUTING.md:13]:
```suggestion
- 3.1 In the Description section add everything that can help the reviewer and the reviewing process, like a description of the issue and the way you decided to go about it, screenshots or gifs, etc.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+When contributing to this repository, please first discuss the change you wish to make via [issue](https://github.com/wix/react-native-ui-lib/issues/new/choose),
+[Discord channel](https://discord.gg/2eW4g6Z), or any other method with the owners of this repository before making a change. 
+
+## Pull Request Process
+
+1. First make sure the environment is working and synced with master. For installation details go [here](https://github.com/wix/react-native-ui-lib/blob/master/markdowns/getting-started/setup.md#demo-app)
+2. Before submitting a PR we suggest running `npm run prepush` command that verifies our lint, TS and tests were not broken.
+3. Please don't change our PR template.
+- 3.1 In Description add everything that can help the reviewer and help the reviewing process. 
+- 3.2 In the Changelog it's important a message for the library users - this message will be included in our release notes
```
Comment (Inbal-Tish) [inline CONTRIBUTING.md:14]:
```suggestion
- 3.2 The Changelog section is important. Write a message for the library users - this message will be included in our release notes.
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+4. We try to make the PR review process as quick as possible, please be patient
+
+
+### Code Standards
+
+- Readability and clean code above all - we believe a clean code is easier to maintain and refactor when needed. 
+- Deciding on an API is not trivial but we always aim to keep it generic and intuitive - feel free to consult with us.
+- Our documentation is derived from our components' prop comments, make sure to add a clear description when adding new props. 
+- When possible, consider adding tets for a new functionality you add.
```
Comment (Inbal-Tish) [inline CONTRIBUTING.md:23]:
```suggestion
- When possible, consider adding tests for the new functionality you added.
```

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+
+
+## Typescript Status
+Currently most of our code is migrated to typescript though we still have some leftovers of javascript code and manual typings.  
+Till we complete the migration you require to do the following for new TS files
```
Comment (Inbal-Tish) [inline CONTRIBUTING.md:29]:
```suggestion
Until we complete the migration you are required to do the following for new TS files
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+
+## Pull Request Process
+
+1. First make sure the environment is working and synced with master. For installation details go [here](https://github.com/wix/react-native-ui-lib/blob/master/markdowns/getting-started/setup.md#demo-app)
+2. Before submitting a PR we suggest running `npm run prepush` command that verifies our lint, TS and tests were not broken.
+3. Please don't change our PR template.
+- 3.1 In Description add everything that can help the reviewer and help the reviewing process. 
+- 3.2 In the Changelog it's important a message for the library users - this message will be included in our release notes
+4. We try to make the PR review process as quick as possible, please be patient
+
```
Comment (Inbal-Tish) [inline CONTRIBUTING.md:16]:
```suggestion
5. Please don't submit automatic formatting changes to the files as it results in cumbersome code reviewing.
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,31 @@
+# Contributing
+If you got here it means you want to dedicate some of your time and help us with making this library better.  
+So first, thank you ❤️
+
+When contributing to this repository, please first discuss the change you wish to make via [issue](https://github.com/wix/react-native-ui-lib/issues/new/choose),
+[Discord channel](https://discord.gg/2eW4g6Z), or any other method with the owners of this repository before making a change. 
+
+## Pull Request Process
```
Comment (ethanshar) [inline CONTRIBUTING.md:8]: Also worth mentioning PR prefixes (fix, feat, etc..)

File: CONTRIBUTING.md (hunk 8 above)
Comment (ethanshar) [reply CONTRIBUTING.md:16]: But we do want them to keep the code formatted according to our guidelines.. no?

File: CONTRIBUTING.md (hunk 1 above)
Comment (ethanshar) [reply CONTRIBUTING.md:32]: 1. done, added another section in Code Standards 
2. not sure I understand, is this a question? or a suggestion to rephrase the sentence?  (: 
3. Added a bullet for that, thank

Comment (mendyEdri) [reply CONTRIBUTING.md:32]: Great :)
2. it's now more clear you referring to unit tests (`..tests for the new functionality you added`)

File: CONTRIBUTING.md (hunk 8 above)
Comment (Inbal-Tish) [reply CONTRIBUTING.md:17]: Yes. but if they auto format it will take their environment's settings, not ours

Comment (Inbal-Tish) [reply CONTRIBUTING.md:17]: I'm not sure that works... If the user has some other formatting tool configured in his environment, wouldn't it override the project's format configuration?

Comment (ethanshar) [reply CONTRIBUTING.md:17]: Maybe, I think we'll learn more after a few PRs and see how it goes. 
And then we can add a more relevant message here about formatting..

Comment (Inbal-Tish) [reply CONTRIBUTING.md:17]: Sound good

**Pull request merged**